pkg/providers/aws/provider_redis_metrics.go

```diff
@@ -102,58 +90,100 @@ func (r *RedisMetricsProvider) scrapeRedisCloudWatchMetricData(ctx context.Conte
 	// for more info see https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_GetMetricData.html
 	logger := resources.NewActionLogger(r.Logger, "scrapeRedisCloudWatchMetricData")
 	logger.Infof("scraping redis instance %s cloud watch metrics", resourceID)
-	metricOutput, err := cloudWatchApi.GetMetricData(&cloudwatch.GetMetricDataInput{
-		// build metric data query array from `elasticacheCloudWatchMetricTypes`
-		MetricDataQueries: buildRedisMetricDataQuery(resourceID),
-		// metrics gathered from start time to end time
-		StartTime: aws.Time(time.Now().Add(-resources.GetMetricReconcileTimeOrDefault(resources.MetricsWatchDuration))),
-		EndTime:   aws.Time(time.Now()),
-	})
-	if err != nil {
-		return nil, errorUtil.Wrap(err, "error getting metric for elasticache")
-	}
 
-	// get cluster if for use in metric labels
-	clusterID, err := resources.GetClusterID(ctx, r.Client)
-	if err != nil {
-		return nil, errorUtil.Wrap(err, "error getting clusterID")
+	var metrics []*providers.GenericCloudMetric
+	listOutput, err := elastiCacheApi.DescribeReplicationGroups(&elasticache.DescribeReplicationGroupsInput{})
+    if err != nil {
+    	logrus.Error("failed to describe replicationGroups",err)
 	}
-
-	// ensure metric data results are not nil
-	if metricOutput.MetricDataResults == nil {
-		return nil, errorUtil.New("no metric data returned from elasticache cloudwatch")
+	replicationGroups := listOutput.ReplicationGroups
+	// Metrics are returned per node for ElastiCache
+	var foundCache *elasticache.ReplicationGroup
+	for _, c := range replicationGroups {
+		if *c.ReplicationGroupId == resourceID {
+			foundCache = c
+			break
+		}
 	}
+	// poll MemberCluster array for CacheClusterId
+	for _, cacheClusterId := range foundCache.MemberClusters {
+		//cast string pointer to string
+		cacheClusterIdString := "" + *cacheClusterId
+		metricOutput, err := cloudWatchApi.GetMetricData(&cloudwatch.GetMetricDataInput{
+			// build metric data query array from `metricType`
+			MetricDataQueries: buildRedisMetricDataQuery(cacheClusterIdString, metricTypes),
+			// metrics gathered from start time to end time
+			StartTime: aws.Time(time.Now().Add(-resources.GetMetricReconcileTimeOrDefault(resources.MetricsWatchDuration))),
+			EndTime:   aws.Time(time.Now()),
+		})
+		if err != nil {
+			return nil, errorUtil.Wrap(err, "error getting metric for elasticache")
```

Comment:
Since we are looping here through the two cache clusters, do we want to return if one of them errors. 

It might be worth logging the error and running with a `continue` instead of a return. So that we ensure we expose all metrics regardless if one of them fails.

Reply:
good point